### PR TITLE
Fix issues on PR 1244

### PR DESCRIPTION
**Pull request by bdilly**
I've fixed a few issues in a commit on top of
PR #1244 commits

Signed-off-by: Bruno Dilly <bruno.dilly@intel.com>

### CODE REVIEW

File: doc/doxygen/Doxyfile [hunk 1]
```diff
@@ -272,7 +272,7 @@ EXPAND_ONLY_PREDEF     = YES
 SEARCH_INCLUDES        = YES
 INCLUDE_PATH           =
 INCLUDE_FILE_PATTERNS  =
-PREDEFINED             = __attribute__((x))=
+PREDEFINED             = __attribute__((x))=  DOXYGEN_RUN
```
Comment (glima) [inline doc/doxygen/Doxyfile]: I ran doxygen a couple of times and never saw this being set. Beats me when it's supposed to be set on to what.

File: src/lib/common/include/sol-atomic.h [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+/* Order levels */
+#define SOL_ATOMIC_RELAXED memory_order_relaxed
+#define SOL_ATOMIC_CONSUME memory_order_consume
+#define SOL_ATOMIC_ACQUIRE memory_order_acquire
+#define SOL_ATOMIC_RELEASE memory_order_release
+#define SOL_ATOMIC_ACQ_REL memory_order_acq_rel
+#define SOL_ATOMIC_SEQ_CST memory_order_seq_cst
+
+/* Types */
+typedef atomic_int sol_atomic_int;
```
Comment (glima) [inline src/lib/common/include/sol-atomic.h]: We got atomic_uchar but I guess it's no big deal to fallback to int on boolean atomics.

File: src/lib/common/sol-worker-thread-impl-riot.c [hunk 3]
```diff
@@ -137,7 +137,7 @@ sol_worker_thread_do(void *data)
     /* From this point forward, we can't allow a context switch. IRQ will be
      * re-enabled by the scheduler once this function returns */
     irq_disable();
-    sol_atomic_store(&thread->finished, true, SOL_ATOMIC_SEQ_CST);
+    sol_atomic_store(&thread->finished, true, SOL_ATOMIC_RELEASE);
```
Comment (glima) [inline src/lib/common/sol-worker-thread-impl-riot.c]: I can agree we do not require ordering there, but I hope there are no pitfalls.

File: doc/doxygen/Doxyfile (hunk 1 above)
Comment (bdilly) [inline doc/doxygen/Doxyfile]: Indeed I don't think it's having any effect in our project

I'll drop it.

**Pull request merged**